neo4j-cypher-dsl/src/main/java/org/neo4j/cypherdsl/core/Node.java

```diff
@@ -63,7 +63,7 @@ static Node create() {
 		return new Node(null, null);
 	}
 
-	private final SymbolicName symbolicName;
+	private volatile SymbolicName symbolicName;
```

Comment:
I think this field could still remain final, and just be assigned to an unresolved SymbolicName instead of 'null' on line 74. It would have the same outcome but wouldn't require the synchronization, and could keep this field final. The only downside is an unresolved symbolic name object may be created sometimes when it is not necessary, but that seems like less of a concern than having to synchronize at other points

Reply:
I wanted to do this approach first, but then we would generate a couple of different queries which are not _requiring_ a symbolic name.